packages/auditd_manager/data_stream/auditd/agent/stream/auditd.yml.hbs

```
@@ -7,8 +7,17 @@ socket_type: '{{socket_type}}'
 immutable: {{immutable}}
 resolve_ids: {{resolve_ids}}
 failure_mode: {{failure_mode}}
-{{#if audit_rules}}
+{{#if session_data}}
+audit_rules: "{{escape_multiline_string audit_rules}}
```

I think you'll need to rework this to be like `processors` below so you can have both `session_data` and `audit_rules` and only one `audit_rules` block

I also got an `escape_multiline_string`, but I maybe didn't have the needed kibana changes

---

I could manage to have processors logic cleaner because it uses type yaml, so the UI validates the Yaml format, audit_rules uses textarea and its more flexible in the input, but trickier to concat.

> I also got an escape_multiline_string, but I maybe didn't have the needed kibana changes

Yes that was merged today on kibana main so it will probably be available in the next snapshot.

---

Could you have `audit_rules:` outside the if statements?

It's ok to have an empty `audit_rules` block, but not have two `audit_rules` (auditbeat won't report any error, but the 2nd block just won't work, so it can cause a silent error), and it's important to be able to have both the session data rules and any other custom rules users add 

I haven't tried it, but maybe something like this will work:
```
audit_rules: |
    {{#if session_data}}
    "{{escape_multiline_string audit_rules}}
      {{escape_multiline_string "
      # Session data audit rules
      -a always,exit -F arch=b64 -S execve,execveat -k exec
      -a always,exit -F arch=b64 -S exit_group
      -a always,exit -F arch=b64 -S setsid"}}"
    {{/if}}
    {{#if audit_rules}}
    {{escape_string audit_rules}}
    {{/if}}
```

---

I see what you mean, I tried your solution but I got this `YAMLException`

```
YAMLException: can not read a block mapping entry; a multiline key may not be an implicit key (22:7)

 19 | -a always,exit -F arch=b64 -S open,tr ...
 20 |
 21 |
 22 |       # Session data audit rules
------------^
 23 |
 24 |       -a always,exit -F arch=b64 -S e ...
```

However, I adjusted the indentation to make sure once the handlebar file is parsed it doesn't include the key twice:

Those are the streams part of the `elastic-agent.yaml` for each scenario:


### Session data checked, Audit rules empty:

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-c430e68e-fcde-4317-90f8-d5bccaf25e75
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: |-


          # Session data audit rules
          -a always,exit -F arch=b64 -S execve,execveat -k exec
          -a always,exit -F arch=b64 -S exit_group
          -a always,exit -F arch=b64 -S setsid
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
        processors:
          - add_session_metadata:
              backend: auto
```

### Audit rules filled, Session data unchecked

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-5bc9f711-81fd-47cc-8bae-dbb44b9f981f
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: >-
          # Unauthorized access attempts to files (unsuccessful).

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
```

### Session data checked, audit rules filled

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-62408e1a-806d-406f-aba8-0ae1506f15f6
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        audit_rules: >-
          # Unauthorized access attempts to files (unsuccessful).

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b32 -S
          open,creat,truncate,ftruncate,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EACCES
          -F auid>=1000 -F auid!=4294967295 -F key=access

          -a always,exit -F arch=b64 -S
          open,truncate,ftruncate,creat,openat,open_by_handle_at -F exit=-EPERM
          -F auid>=1000 -F auid!=4294967295 -F key=access


          # Session data audit rules

          -a always,exit -F arch=b64 -S execve,execveat -k exec

          -a always,exit -F arch=b64 -S exit_group

          -a always,exit -F arch=b64 -S setsid
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd
        processors:
          - add_session_metadata:
              backend: auto
```

### session unchecked, audit rules 

```yaml
streams:
      - id: >-
          audit/auditd-auditd_manager.auditd-d17efb9a-8968-49b7-b5a9-8a8377611659
        type: audit/auditd
        data_stream:
          dataset: auditd_manager.auditd
          type: logs
        condition: ${host.platform} == 'linux'
        include_raw_message: true
        socket_type: ''
        immutable: false
        resolve_ids: true
        failure_mode: silent
        backlog_limit: 8192
        rate_limit: 0
        include_warnings: false
        backpressure_strategy: auto
        tags:
          - auditd_manager-auditd

```

It should be working as expected now, no duplicated `audit_rules` in the yaml

---

Thanks, that looks good to me, and the config in the agent looks good